test/marginal/test_selection_variable_scikit_learn_classification.py

```diff
@@ -0,0 +1,342 @@
+    ):
+        super().__init__(
+            problem_type="classification",
+            discrete_features=discrete_features,
+            n_neighbors=n_neighbors,
+            random_state=random_state,
+            n_jobs=n_jobs,
+        )
+
+
+def classification_float(y, nb_classes=4, min_value=None, max_value=None):
+    """
+    Create classification problem bae on regression problem
+    Parameters
+    ----------
+    y : array-like of shape (n_samples,)
+        target for the prediction
+    nb_classes : int, default=4
+        number of classes
+    min_value : None or int, default=None
+        maximal value for the first class
+    max_value : None or int, default=None
+        minimal value for the last class
+    Returns
+    -------
+    array-like of shape (n_samples,)
+        classification for target
+    """
+    assert nb_classes >= 2
+    if min_value is None:
+        min_value = np.min(y)
+    if max_value is None:
+        max_value = np.max(y)
+    assert min_value < max_value
+    # change from regression to classification problem
+    y_ = deepcopy(y)
+    previous_value = min_value
+    for classe, value in enumerate(np.linspace(min_value, max_value, nb_classes)):
+        if value == min_value:
+            y_[np.where(y < min_value)] = classe
+        elif value == max_value:
+            y_[np.where(y >= max_value)] = classe
+        else:
+            y_[np.where(np.logical_and(previous_value <= y, y < value))] = classe
+        previous_value = value
+    y_ = np.array(y_, dtype=int)
+    return y_
+
+
+def configure_marginal_classication(
+    ClassMethod, X, y, nb_classes=4, min_value=None, max_value=None
+):
+    """
+    Configure ClassMethod model for feature importance analysis.
+    Parameters
+    ----------
+    X : array-like of shape (n_samples, n_features)
+        Input data matrix where each column represents a feature
+        and each row a sample.
+    y : array-like of shape (n_samples,)
+        Target variable array.
+    Returns
+    -------
+    importance : array-like
+        Array containing importance scores for each feature.
+        Higher values indicate greater feature importance in predicting
+        the target variable.
+    Notes
+    -----
+    The function performs the following steps:
+    1. Intanciate ClassMethod
+    2. Calculates feature importance
+    """
+    y_ = classification_float(y, nb_classes, min_value, max_value)
+    # instantiate model
+    vi = ClassMethod()
+    # fit the model using the training set
+    vi.fit()
+    # calculate feature importance using the test set
+    importance = vi.importance(X, y_)
+    return np.array(importance)
+
+
+parameter_exact = [
+    ("HiDim", 150, 200, 1, 0.0, 42, 1.0, np.inf, 0.0),
+    ("HiDim with noise", 150, 200, 1, 0.0, 42, 1.0, 10.0, 0.0),
+    ("HiDim with correlated noise", 150, 200, 1, 0.0, 42, 1.0, 10.0, 0.5),
+    ("HiDim with correlated features", 150, 200, 1, 0.8, 42, 1.0, np.inf, 0.0),
+]
```

Comment:
For a univariate selection, it is not necessary to create such a high-dimensional example. especially if the support size is only one. `n_features` could be lowered to spare compute.

Reply:
Do you mean that univariate can't detect features in high dimensions?

Reply:
In the case of univariate selection, features are handled randomly, so there is no methodological difference between low- and high- dimension. We can simply the tests here.

Reply:
see issue #375.